pkg/install/resources.go

```diff
@@ -217,8 +217,10 @@ func AllResources(o *VeleroOptions) (*unstructured.UnstructuredList, error) {
 	sa := ServiceAccount(o.Namespace)
 	appendUnstructured(resources, sa)
 
-	sec := Secret(o.Namespace, o.SecretData)
-	appendUnstructured(resources, sec)
+	if o.SecretData != nil {
```

Comment:
If we're not creating the secret, then we need to remove the secret mount from the Deployment/DaemonSet as well (see `WithoutCredentialsVolume`). It would also be nice to not set the credentials environment variables in this case